src/main/java/org/snomed/snowstorm/validation/ConceptValidationThenOperationService.java

```diff
@@ -0,0 +1,223 @@
+package org.snomed.snowstorm.validation;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.ihtsdo.drools.response.InvalidContent;
+import org.ihtsdo.drools.response.Severity;
+import org.snomed.snowstorm.core.data.domain.Axiom;
+import org.snomed.snowstorm.core.data.domain.Concept;
+import org.snomed.snowstorm.core.data.domain.ConceptView;
+import org.snomed.snowstorm.core.data.domain.Description;
+import org.snomed.snowstorm.core.data.domain.Relationship;
+import org.snomed.snowstorm.core.data.domain.SnomedComponent;
+import org.snomed.snowstorm.core.data.services.ConceptService;
+import org.snomed.snowstorm.core.data.services.ServiceException;
+import org.snomed.snowstorm.rest.ControllerHelper;
+import org.snomed.snowstorm.validation.domain.DroolsDescription;
+import org.snomed.snowstorm.validation.domain.DroolsRelationship;
+import org.snomed.snowstorm.validation.exception.DroolsValidationException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Objects;
+import java.util.Random;
+import java.util.Set;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+
+import static org.snomed.snowstorm.rest.ControllerHelper.getCreatedLocationHeaders;
+
+@Service
+public class ConceptValidationThenOperationService {
```

Comment:
I feel like this should be a helper class rather than a service class as it deals with some controller concerns rather than just business logic.
If we have ConceptValidationHelper with a method like `validate` that could run the validation and throw an exception if there are errors or return a validation header if there are warnings.
That way the final call to the ConceptService to create or update and the creation of the `ResponseEntity` could be in the `ConceptController` which would make for easy reading and keep controller concerns in one place as far as possible. 
This is a suggestion, what are your thoughts?

Reply:
Yes I agree this would make it more readable and will make the code changes in a separate commit (while I'll do now). Thanks.

Reply:
Fixed.